### DISCUSSION TITLE
Specifying parameters in openapi_extras overrides query parameters, rather than appending them

### DISCUSSION THREAD

**Discussion by orilevari** [fastapi/fastapi#6467] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

See the PR I opened with a fix: https://github.com/tiangolo/fastapi/pull/4267 It includes a test case that describes the use case.

### Description

When specifying parameters in openapi_extras for an endpoint, they will overwrite the existing query parameters with the extra parameters, rather than appending the parameters to the spec. This is due to deep_dict_update in utils.py overwriting list types rather than merging them together.

I have created a PR already which has a fix and adds a test case: https://github.com/tiangolo/fastapi/pull/4267
These updates change the behavior to merge parameters together rather than overwriting.

If y'all agree that this seems like a bug, I'd appreciate a review on the PR. Thanks in advance! 😄 

### Operating System

macOS

### FastAPI Version

0.70.0

### Python Version

Python 3.9.7

**Comment by orilevari** [comment, marked as answer]
PR got merged! Closing now

**Comment by tiangolo** [comment]
Thanks for coming back to close it! ☕🍰
